HHS/Head-Start-TTADP
# [TTAHUB-534] add state filter control to filter panel

## Description of change
- added a "State" filter to the filter panel
- added a backend route that gets possible state values based on users regions -> grants

## How to test
- View the filter panel. Check out the state filter and enjoy how it filters by state. Also verify the correct states show in the panel (in my case, there were very few grants with states)

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-534

## Checklists

### Every PR


- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy


### After merge/deploy
